test/e2e/mirror_test.go

```diff
@@ -0,0 +1,160 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	config "github.com/openshift/api/config/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	batch "k8s.io/api/batch/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
+)
+
+const (
+	testMirrorSetName               = "e2e-powershell-mirror"
+	upstreamPowershellImageLocation = "mcr.microsoft.com/powershell"
+	badPowershellImageLocation      = "test.registry.io/powershell"
+)
+
+func testImageMirroring(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	require.NoError(t, tc.loadExistingNodes(), "error getting the current Windows nodes in the cluster")
+
+	if !t.Run("Mirror settings applied to nodes", tc.testMirrorSettingsApplied) {
+		// No point in running the rest of the tests if settings aren't applied
+		return
+	}
```

Comment:
This would be tested by the workload running sucessfully.

Reply:
Should I remove this test altogether? IMO this case adds some value to validate there is the expected number of config files. I can add another entry to the test ITMS object to make it more clear if you prefer

Reply:
> IMO this case adds some value to validate there is the expected number of config files

The expected config files will be tested through the workloads successfully running.
If you are saying you are trying to test that there are not more files than expected, that would be better tested within unit tests.